### ISSUE TITLE
[Request] Feature: Add support for Nexus Mods API to update Mods without opening a web browser

### ISSUE THREAD

**Issue by MN4444** [RyanWalpoleEnterprises/The-Potato-Farm-SDVM#119]
Title says it all, just ask the user for their personal API key to use this feature of the mod manager. It would make it much more convenient for the user to update their mods.

**Comment by RyanWalpole** [comment]
> Title says it all, just ask the user for their personal API key to use this feature of the mod manager. It would make it much more convenient for the user to update their mods.

Apologies for the delay, this project has not been on the front runner for a little while as I had added most of the features I'd wanted and was lacking time and motivation to implement much else. I'll be looking at this one for a future update!

**Comment by villainous-j** [comment]
maybe a little blue grass might motivate you 😉

**Comment by RyanWalpole** [comment]
I've been working on an update to the mod manager (selfishly, when I'm about to start a new modded multiplayer save with my friends, I get re-inspired to touch this program). This is definitely on the radar as a feature, though it might take some time for me to implement. 

Since this application started as a tool to help my friends get started with mods, it was never written efficiently or (putting it bluntly) very well at all, so there's a lot of code that needs cleanup before I can start adding more complex features.

TLDR: I'm gonna look at adding this when I can, the blue grass in fact did motivate me. Thanks for your patience while I work on this :)

CC:
@MN4444 
@villainous-j

**Comment by villainous-j** [comment]
> I've been working on an update to the mod manager (selfishly, when I'm about to start a new modded multiplayer save with my friends, I get re-inspired to touch this program). This is definitely on the radar as a feature, though it might take some time for me to implement. Since this application started as a tool to help my friends get started with mods, it was never written efficiently or (putting it bluntly) very well at all, so there's a lot of code that needs cleanup before I can start adding more complex features.

hah! as a developer myself i can understand this very much.

> TLDR: I'm gonna look at adding this when I can, the blue grass in fact did motivate me. Thanks for your patience while I work on this :)

very good very good, please no stress and don't let it take you away too far from the farm.

**Comment by RyanWalpole** [comment]
> hah! as a developer myself i can understand this very much.

Ah, if it isn't the consequences of my own actions! Rapid app development, am I right?

> very good very good, please no stress and don't let it take you away too far from the farm.

With the exception of the Stardew Valley Orchestral Concert next week, I'm free of Stardew until my next modded save starts in about a month (if I'm able to get everyone to agree on a schedule by then!).

I'm pushing out a bug fix update soon and then I'll have time to work on some code cleanup and new features <3 Thanks for your patience again!

**Comment by RyanWalpole** [comment]
Really happy and proud to say I was able to build this feature into the new update, version 240701. It's currently in testing and should be rolling out in the next 24 hours 🤞🏼 if all goes well.

For new users, setting up their Nexus API details will be handled as a part of the first run setup. For existing users, it will be handled through the Settings tab. Please find below some basic instructions on how to set it up.

![image](https://github.com/user-attachments/assets/8fcb1bbf-eaf5-4679-97be-f8e153944d07)

Update Check             |  Update Download
:-------------------------:|:-------------------------:
![updatecheck](https://github.com/user-attachments/assets/045128cf-8bde-4f09-9adf-8fd6e6696ce7)  |  ![updateapply](https://github.com/user-attachments/assets/3928fa6b-6511-47ee-86d8-3445e3e2ea80)


Thanks for your suggestion and I hope that it's as bug free for you as it has been for me 😅 
Keep an eye out for the update in the coming 24 hours or so and happy playing!

CC:
@villainous-j 
@MN4444

**Comment by RyanWalpole** [comment]
This is officially released. Going to close this issue. Please open another bug report for any bugs pertaining to this new feature.

**Comment by villainous-j** [comment]
thank you so much for putting that together! it looks awesome. when i get back to the valley (which is inevitable) i very much look forward to using this!
